src/components/controllers/slot-controller.ts

```diff
@@ -132,59 +134,80 @@ export class SlotController implements ReactiveController {
    * @todo: Continue improvement and performance in this method.
    * @todo: Observe slot changes and adding new elements to the slot.
    *
-   * @param slotName string The name of the named slot.
+   * @param slotShadowDom The named slot to be moved.
    */
-  private moveNamedSlot(slotName: string) {
-    // Reassign host for better typing.
-    const host = this.host as unknown as ReactiveElement;
-
-    const shadowSlotLocation = host.renderRoot
-      ? host.renderRoot.querySelector(`slot[name="${slotName}"]`)
-      : false;
-
-    if (shadowSlotLocation) {
-      const slotLightDom = host.querySelector('[slot=' + slotName + ']');
-
-      // Only if named slot found in light dom - move it into ShadowDOM.
-      if (slotLightDom) {
-        shadowSlotLocation.append(slotLightDom);
-      }
-    }
+  private moveNamedSlots(slotShadowDom: HTMLSlotElement) {
+    const slotLightDomArray = this.hostEl.querySelectorAll(
+      `:scope > [slot="${slotShadowDom.name}"]`
```

Comment:
It looks as though this line limits this logic to slots that are a direct child of the host element. So the implication would be that this would not apply to a slot wrapped in a <div>, for example. Is that the intention, or am I misunderstanding something here?

Reply:
The idea is to avoid affecting slots from child web components. See https://github.com/phase2/outline/pull/304 for a demonstration. If you have nested components with the same slot name, you could get elements from the wrong element when you do this query without `:scope >`.

`this.hostEl` is the components light DOM. If you have slots with web components in them, they should handle their own behavior.

I could be wrong, but I believe for slots to behave as expected, they need to be direct children of the light DOM component. Is there a use case for wrapping slots in other elements and having them function as slots on the component?

Reply:
OK, I think I understand how this works, now. To paraphrase, the selector is actually pointing to the slotted Light DOM node, which is always a direct child of the host element, irrespective of where it then gets dropped into the Shadow DOM tree.

Thanks for the detailed explanations, @damontgomery and @shaal !!

Reply:
Yes, I believe you've got it. I get mixed up with all these DOMs, so no worries. :)